op-challenger/game/scheduler/coordinator_test.go

```diff
@@ -182,6 +181,57 @@ func TestDeleteDataForResolvedGames(t *testing.T) {
 	require.True(t, disk.gameDirExists[gameAddr3], "game 3 data should be preserved (inflight)")
 }
 
+func TestSchedule_RecordActedL1Block(t *testing.T) {
+	c, workQueue, _, _, _ := setupCoordinatorTest(t, 10)
+	gameAddr3 := common.Address{0xcc}
+	ctx := context.Background()
+
+	// The first game should be tracked
+	require.NoError(t, c.schedule(ctx, asGames(gameAddr3), 1))
+	require.Equal(t, uint64(1), c.blockUpdates[1])
+
+	// Process the result
+	require.Len(t, workQueue, 1)
+	j := <-workQueue
+	j.block = 1
+	j.status = types.GameStatusDefenderWon
+	require.NoError(t, c.processResult(j))
+
+	// Verify that the block number is recorded by the metricer as acted upon
+	require.Equal(t, uint64(0), c.blockUpdates[1])
+	require.Equal(t, uint64(1), c.m.(*stubSchedulerMetrics).actedL1Blocks)
+}
+
+func TestSchedule_RecordActedL1BlockMultipleGames(t *testing.T) {
+	c, workQueue, _, _, _ := setupCoordinatorTest(t, 10)
+	gameAddr1 := common.Address{0xaa}
+	gameAddr2 := common.Address{0xbb}
+	gameAddr3 := common.Address{0xcc}
+	ctx := context.Background()
+
+	games := asGames(gameAddr1, gameAddr2, gameAddr3)
+	require.NoError(t, c.schedule(ctx, games, 1))
+	require.Equal(t, uint64(3), c.blockUpdates[1])
+	require.Len(t, workQueue, 3)
+
+	// Game 1 progresses and is still in progress
+	// Game 2 progresses and is now resolved
+	// Game 3 hasn't yet progressed (update is still in flight)
+	for i := 0; i < len(games); i++ {
+		require.Equal(t, uint64(0), c.m.(*stubSchedulerMetrics).actedL1Blocks)
+		j := <-workQueue
+		if j.addr == gameAddr2 {
+			j.status = types.GameStatusDefenderWon
+		}
+		j.block = 1
```

Comment:
We shouldn't be setting the job block number in any of these tests - it should be set correctly by the scheduler when creating the job.